beater/http.go

```diff
@@ -85,24 +114,25 @@ func newHTTPServer(
 		MaxHeaderBytes: cfg.MaxHeaderSize,
 	}
 
+	// TODO: Track TLS settings, if it changes then we need to restart the
+	// server
```

Comment:
I'm not sure if we need to restart the server or not when changing tls settings in the server. Since we `accept()` on the socket, maybe the tls handshake is done with the current tls config on the server and not a copy of the initial config when starting `server.Serve(listener)`

Reply:
in `httpServer.start()` there's a branch between `ServeTLS()` and `Serve()`, so turning tls on or off will require a restart, but modifying settings may not.